vispy/visuals/graphs/util.py

```diff
@@ -0,0 +1,103 @@
+# -*- coding: utf-8 -*-
+# Copyright (c) 2015, Vispy Development Team.
+# Distributed under the (new) BSD License. See LICENSE.txt for more info.
+"""
+Graph utilities
+===============
+
+A module containing several graph utility functions.
+"""
+
+import itertools
+
+import numpy as np
+
+
+def straight_line_vertices(adjacency_mat, node_coords, directed=False):
+    """
+    Generate the vertices for straight lines between nodes.
+
+    If it is a directed graph, it also generates the vertices which can be
+    passed to an :class:`ArrowVisual`.
+
+    Parameters
+    ----------
+    adjacency_mat : array
+        The adjacency matrix of the graph
+    node_coords : array
+        The current coordinates of all nodes in the graph
+    directed : bool
+        Wether the graph is directed. If this is true it will also generate
+        the vertices for arrows which can be passed to :class:`ArrowVisual`.
+
+    Returns
+    -------
+    vertices : tuple
+        Returns a tuple containing containing (`line_vertices`,
+        `arrow_vertices`)
+    """
+    if adjacency_mat.shape[0] != adjacency_mat.shape[1]:
```

Comment:
`if adjacency_mat.ndim !=2 or ...`

We should also type-check. You might want to put this at the top of the file:
```
try:
    from scipy.sparse import issparse
except ImportError:
    def issparse(x):
        return False
```
Then your type-check here can basically be an attempt to cast it to the proper type before dimension checking:
```
if not issparse(adjacency_mat)) and not isinstance(adjacency_mat, ndarray):
    adjacency_mat = np.array(adjacency_mat, float)
```